### ISSUE TITLE
"Delete File" / "Delete Directory" confirmation dialogue doesn't respond to mouse input

### ISSUE THREAD

**Issue by lorenzlm** [Zettlr/Zettlr#1645]
## Description

When you use the context menu (right click on title in the File Manager) or the menu item to **delete a file or directory,** the **modal popup confirmation dialog** does not respond to mouse click. "Enter" to confirm and "Esc" to cancel works, though.

## Reproducing

Install Zettlr 1.8.6 (although, as far as I remember, the bug was already present in 1.8.5)

## Expected behaviour

The modal popup confirmation dialog should also work with mouse.

## Platform
 - OS and version: Linux Mint 20.1 Ulyssa (Cinnamon Desktop)
 - Zettlr Version: 1.8.6

**Comment by zargex** [comment]
This happens to me too.

Platform:
Debian 10.7
Zettlr 1.8.5

**Comment by crbald** [comment]
It works just fine for me for files and directories.

Platform: 
Windows 10 1909
Zettlr 1.8.6

Could it be a Linux issue? Or does it also happen for Mac people?

**Comment by Cmhorn** [comment]
This occurs with me as well under the same conditions. 

PopOs 20.10
Zettlr 1.8.6

**Comment by ploum** [comment]
Same for me on Ubuntu with Zettlr 1.8.5 and 1.8.6

**Comment by nathanlesage** [comment]
I am very sure this is some bug in the underlying Electron framework, because both Windows and macOS users report that it's working just fine (as with other dialog types). There are some other issues for the other kinds of dialogs Zettlr has open here, but there some people are skeptic that it is a bug with the underlying framework …

**Comment by jchalifour** [comment]
This also happens on Manjaro (Gnome 3.38.3) with Zettlr 1.8.6 (and I think I remember it being the case for 1.8.5). However, I noticed that a similar sort of dialogue does allow me to click on it. For example, if I create a file, then delete it, that entry still shows up in the Recent Documents portion of the File menu. If I click its name there, a dialogue pops up telling me that it can't find the file of course but it does allow me to click it with the mouse. I don't know whether this is helpful or not... thought it might be a clue.

![zettlr-dialogues_clickable-notclickable_20210120](https://user-images.githubusercontent.com/5346551/105279681-1914e380-5b76-11eb-8812-e294f822df7b.png)

**Comment by zargex** [comment]
Hi, I was just testing previous version (only the AppImage) and the delete file confirmation is clickable in:

-  Zettlr-1.5.0-x86_64.AppImage
-  Zettlr-1.6.0-x86_64.AppImage
-  Zettlr-1.7.0-x86_64.AppImage

But not in

-  Zettlr-1.8.0-x86_64.AppImage

I guess something changed from 1.7 to 1.8

**Comment by zargex** [comment]
Hi, I just checked again in which version this behaviour started, and looks like it was in **1.8.0-beta.2** (beta.1 has a bug that doesn't show the confirmation window). 
Also I want to share the difference between **1.8.0-beta.2**
![1 8 0-beta 2](https://user-images.githubusercontent.com/9205985/106678538-f7a9f380-6588-11eb-9f57-888c37957bde.png)

 and **1.8.0-beta**
![1 8 0-beta](https://user-images.githubusercontent.com/9205985/106678553-fd073e00-6588-11eb-9a52-f4a7f272acad.png)

The small icon in the top left corner changed (and that still in 1.8.5). Not saying that is important or anything but maybe is related to this bug.

**Comment by nathanlesage** [comment]
> The small icon in the top left corner changed (and that still in 1.8.5). Not saying that is important or anything but maybe is related to this bug.

That is interesting – can you determine which Electron version is used in both? (Should be listed in the About dialog, in the last tab) This would pinpoint the exact version where this behaviour started.

**Comment by zargex** [comment]
- 1.8.0-beta: Electron v10.1.4
- 1.8.0-beta.2: Electron v10.1.5

**Comment by nathanlesage** [comment]
Thank you. I checked the Electron changelog and while they touched the dialog functionality, nothing in there changed for Linux operating systems.

I did some more research and indeed, [between 1.8 beta and beta.2](https://github.com/Zettlr/Zettlr/compare/1.8.0-beta...1.8.0-beta.2) I actually introduced the window manager which also handles the dialogs. And I know that I fixed quite some things in the dialogs, most importantly, to either attach or not attach them to an open window. There the error might lay.

But I can't find any error after [double checking the corresponding docs](https://www.electronjs.org/docs/api/dialog#dialogshowmessageboxbrowserwindow-options). Anyone else has an idea?

**Comment by maweki** [comment]
The common issue seems to be Gnome, right? But not all dialogues are affected.

Affected:
- Open/Open Directory from the Menu
- Import Translations from the Menu
- Delete File **from the Sidebar** 

Not Affected:
- Open Directory **from the Sidebar**
- Open Workspace from the quicklaunch menu (icon bar)
- Open File from Strg+o
- Select Bib-file from the Settings Dialog

Everything from the menu seems to not work. The stuff from the sidebar is inconsistent.

Everything is fixed by moving/changing size of the window in the background.

**Comment by nathanlesage** [comment]
So it's neither alone the menu nor alone the GUI. Which explains why I couldn't pinpoint any logical error in my code thus far

**Comment by maweki** [comment]
I think this only happens when it comes from a context menu. Either right-clicking or the menu button open a similar context menu/popover/dropdown.

**Comment by nathanlesage** [comment]
You're right, I forgot about that one. Mh, maybe then it'll come back once I re-instate the native menus for 2.0

**Comment by zargex** [comment]
@maweki  I don't think Gnome is the common issue as I use KDE. But I just checked and you are right, moving the window in the background "fix" the issue.

**Comment by zargex** [comment]
I just finished a quick git bisect and seems that the error started with this commit: 
https://github.com/Zettlr/Zettlr/commit/3b728a8dfb24490bbd9a8f7ef3b873db30b7c7d4

**Comment by maweki** [comment]
> I just finished a quick git bisect and seems that the error started with this commit:
> [3b728a8](https://github.com/Zettlr/Zettlr/commit/3b728a8dfb24490bbd9a8f7ef3b873db30b7c7d4)

That looks suspicously like it could cause the issue at hand. Somehow the dialog does not receive any new signals until the background window is moved.

**Comment by nathanlesage** [comment]
That commit, I'm afraid, cannot be the cause, maybe a later one.

However, since the Zettlr codebase is now (a) completely different from just a few weeks ago, and we have moved to Electron 12, if someone would be so kind to test it out locally on a Linux machine to confirm whether or not this bug still occurs, this would help the resolution of this issue incredibly!

**Comment by lduktus** [comment]
I am using Zettlr AppImage 1.8.7. with Fedora 34 and GNOME 40 and can confirm that I seem to have the same issue.

**Comment by yaxu** [comment]
> if someone would be so kind to test it out locally on a Linux machine to confirm whether or not this bug still occurs, this would help the resolution of this issue incredibly!

Do you just need someone to compile+run the develop branch? I can do that..

**Comment by yaxu** [comment]
Unfortunately it isn't running for me:
```shell
alex@teff:~/src/Zettlr$ ~/node_modules/.bin/yarn start
yarn run v1.22.10
$ electron-forge start
✔ Checking your system
✔ Locating Application
✔ Preparing native dependencies
✔ Compiling Main Process Code
✔ Launch Dev Servers
⠇ Compiling Renderer CodeIssues checking in progress...
⠸ Compiling Renderer CodeNo issues found.
⠇ Compiling Renderer CodeIssues checking in progress...
⠹ Compiling Renderer CodeNo issues found.
⠧ Compiling Renderer CodeIssues checking in progress...
⠇ Compiling Renderer CodeIssues checking in progress...
⠴ Compiling Renderer CodeNo issues found.
⠙ Compiling Renderer CodeIssues checking in progress...
⠼ Compiling Renderer CodeIssues checking in progress...
⠦ Compiling Renderer CodeNo issues found.
No issues found.
⠏ Compiling Renderer CodeIssues checking in progress...
⠼ Compiling Renderer CodeNo issues found.
⠏ Compiling Renderer CodeIssues checking in progress...
assets by chunk 3.41 MiB (auxiliary name: about) 81 assets
assets by path win-main/assets/ 230 KiB 3 assets
assets by path main_window/ 10.4 MiB 2 assets
assets by path quicklook/ 8.61 MiB 2 assets
assets by path stats/ 3.64 MiB 2 assets
assets by path defaults/ 3.48 MiB 2 assets
assets by path about/ 3.34 MiB 2 assets
assets by path preferences/ 3.25 MiB 2 assets
assets by path paste_image/ 2.99 MiB 2 assets
assets by path log_viewer/ 2.9 MiB 2 assets
assets by path update/ 2.9 MiB 2 assets
assets by path tag_manager/ 2.89 MiB 2 assets
assets by path error/ 2.86 MiB 2 assets
assets by path print/ 2.86 MiB 2 assets
1 asset
runtime modules 312 KiB 168 modules
orphan modules 443 bytes [orphan] 1 module
modules by path ./node_modules/ 7.07 MiB
  javascript modules 7.04 MiB 1216 modules
  json modules 30 KiB
    ./node_modules/entities/lib/maps/entities.json 28.4 KiB [built] [code generated]
    ./node_modules/entities/lib/maps/legacy.json 1.24 KiB [built] [code generated]
    ./node_modules/entities/lib/maps/xml.json 62 bytes [built] [code generated]
    ./node_modules/entities/lib/maps/decode.json 308 bytes [built] [code generated]
modules by path ./source/ 2.11 MiB 756 modules
11 modules
webpack 5.38.1 compiled successfully in 33127 ms
✔ Compiling Renderer Code
⠋ Compiling Preload ScriptsNo issues found.
⠼ Compiling Preload ScriptsIssues checking in progress...
⠙ Compiling Preload ScriptsIssues checking in progress...
⠙ Compiling Preload ScriptsNo issues found.
⠸ Compiling Preload ScriptsIssues checking in progress...
⠦ Compiling Preload ScriptsNo issues found.
✔ Compiling Preload Scripts
✔ Launching Application


Webpack Output Available: http://localhost:9000

Issues checking in progress...
No issues found.
No issues found.
(node:10984) ExtensionLoadWarning: Warnings loading extension at /home/alex/.config/Zettlr/extensions/nhdogjmejiglipccpnnnanhbledajbpd:
  Unrecognized manifest key 'browser_action'.
  Unrecognized manifest key 'update_url'.
  Permission 'contextMenus' is unknown or URL pattern is malformed.
  Cannot load extension with file or directory name _metadata. Filenames starting with "_" are reserved for use by the system.

(Use `electron --trace-warnings ...` to show where the warning was created)
[12:20:49] こんにちは！ Booting Zettlr at Mon Jun 14 2021 12:20:49 GMT+0100 (British Summer Time).
[12:20:49] Performing environment check ...
[12:20:49] [Application] App is unpackaged, and Pandoc has been found in the resources directory: /home/alex/src/Zettlr/resources/pandoc
[12:20:49] Added DevTools extension:  Vue.js devtools
[12:20:49] Creating required directory /home/alex/.config/Zettlr/defaults ...
[12:20:49] Environment check complete.
[12:20:49] Registering custom protocol safe-file
[12:20:49] Log provider booting up ...
[12:20:49] Config provider booting up ...
[12:20:49] [Config Provider] Loading configuration file from /home/alex/.config/Zettlr/config.json ...
[12:20:49] [Config Provider] Successfully loaded configuration
[12:20:49] Migrating from 1.8.9 to 1.8.8!
[12:20:49] [Config Provider] Migrating Autocorrect replacement --> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement –> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <-- from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <-> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <--> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement ==> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <== from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <=> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <==> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement != from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <> from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement +- from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :times: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :division: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement <= from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement >= from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 1/2 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 1/3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 2/3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 1/4 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 3/4 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 1/8 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 3/8 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 5/8 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement 7/8 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement mm2 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement cm2 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement m2 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement km2 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement mm3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement cm3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement ccm from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement m3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement km3 from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :sup2: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :sup3: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :deg: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :eur: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :gbp: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :yen: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :cent: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement :inr: from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement (c) from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement (tm) from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement (r) from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement ... from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement -- from 'val' to 'value' ...
[12:20:49] [Config Provider] Migrating Autocorrect replacement --- from 'val' to 'value' ...
[12:20:49] Appearance provider booting up ...
[12:20:49] Assets provider starting up ...
[12:20:49] [Assets Provider] Required defaults file export.docx.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.html.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.icml.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.latex.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.odt.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.org.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.pdf.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.plain.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.revealjs.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.rst.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file export.rtf.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.docbook.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.docx.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.epub.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.haddock.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.html.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.latex.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.muse.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.odt.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.opml.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.org.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.rst.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.t2t.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.textile.yaml not found. Copying ...
[12:20:49] [Assets Provider] Required defaults file import.vimwiki.yaml not found. Copying ...
[12:20:49] Citeproc provider booting up ...
[12:20:49] [Citeproc Provider] Loading CSL locale file at /home/alex/src/Zettlr/.webpack/main/assets/csl-locales/locales-en-US.xml ...
[12:20:49] [Citeproc Provider] Loading CSL locale file at /home/alex/src/Zettlr/.webpack/main/assets/csl-locales/locales-en-GB.xml ...
[12:20:49] Dictionary provider booting up ...
[12:20:49] Recent documents provider booting up ...
[12:20:49] Menu provider booting up ...
[12:20:49] Tag provider booting up ...
[12:20:49] Target provider booting up
[12:20:49] CSS provider booting up ...
[12:20:49] Translation provider booting up ...
[12:20:49] Update provider booting up ...
[12:20:49] Notification provider booting up ...
[12:20:49] Stats provider booting up
[12:20:49] [Application] Added /home/alex/src/Zettlr/resources to PATH.
[12:20:49] [Application] The bundled pandoc executable is now in PATH. If you do not want to use the bundled pandoc, uncheck the corresponding setting and reboot the app.
[12:20:49] [Citeproc Provider] Parsing file /home/alex/Dropbox/cite/Alpaca.json as CSL JSON ...
[12:20:49] [Citeproc Provider] Database /home/alex/Dropbox/cite/Alpaca.json loaded (45 items).
[12:20:49] FSAL booting up ...
[12:20:49] Clearing the FSAL cache ...
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type number (-1186530688)
    at validateString (internal/validators.js:124:11)
    at Object.extname (path.js:1229:5)
    at DocumentManager._loadFile (/home/alex/src/Zettlr/.webpack/main/index.js:143495:67)
    at DocumentManager.init (/home/alex/src/Zettlr/.webpack/main/index.js:143411:43)
    at Zettlr.init (/home/alex/src/Zettlr/.webpack/main/index.js:151267:37)
    at /home/alex/src/Zettlr/.webpack/main/index.js:140788:16 {
  code: 'ERR_INVALID_ARG_TYPE'
}
[12:20:49] Could not load URL http://localhost:3000/main_window: ERR_FAILED (-2) loading 'http://localhost:3000/main_window'
Error: ERR_FAILED (-2) loading 'http://localhost:3000/main_window'
    at rejectAndCleanup (electron/js2c/browser_init.js:213:985)
    at Object.stopLoadingListener (electron/js2c/browser_init.js:213:1360)
    at Object.emit (events.js:315:20) {
  errno: -2,
  code: 'ERR_FAILED',
  url: 'http://localhost:3000/main_window'
}
Done in 66.97s.
```

**Comment by nathanlesage** [comment]
@yaxu This might be easily fixable: Apparently there's yet another migration that's missing. Just for starters, would you mind copying your existing `config.json` someplace else (e.g. rename it to `config.backup.json`) and try re-running it? It appears that some of the paths (`activeFile`, I suspect) are stored "wrongly" by the previous version of Zettlr.

P.S.: You should definitely backup the config, because once it's migrated to the 2.0-dev you can't move back :D The config, fortunately, is not that extremely unique …

CC @tobiasdiez: apparently there's more work needed …

**Comment by yaxu** [comment]
@nathanlesage I'm on a different laptop now and I didn't need to move config.json to get it to run. Unfortunately the open dialog still isn't working though. I tried removing the whole of `~/.config/Zettlr` before starting it but that didn't help.

The strange thing is that it thinks it is version 1.8.8, but I am definitely running from the develop branch.

**Comment by yaxu** [comment]
One workaround I've noticed (from the develop branch at least) is that if I open a dialogue, then switch to another virtual desktop and back, I can interact with the dialogue fine. So I guess there is something strange happening with modal focus.

By the way now I'm back on the computer that had the above error, moving the config.json fixed it thanks. But it thinks it's 1.8.8 too.

**Comment by nathanlesage** [comment]
> The strange thing is that it thinks it is version 1.8.8, but I am definitely running from the develop branch.

Yeah, I haven't updated the version number in the develop branch. However, I decided to keep it so that I do not forget to test the updater once before actually beginning to push the 2.0 betas and don't have people complaining because I accidentally broke something again 😂

> So I guess there is something strange happening with modal focus.

Mhhh … I was thinking something similar. I'm really quite unsure what might be the cause because I believe that I implemented these as the docs state. In fact, it's simply calling a function and nothing more, so it might still be a bug in the Electron framework. What contradicts that theory, however, is that it's still there despite many electron apps being in use on Linux.

**Comment by nathanlesage** [comment]
Hey y'all: Would you please mind to try out the nightlies, if the bug is gone there?

You can get them here: https://nightly.zettlr.com

**Comment by ajlewis2** [comment]
I am using Linux Mint 20.1. I tried the nightly and found that the delete file did not work. I installed it by removing the old version and installing the nightly. I tried it with my old configuration folder and also without a configuration folder. Same result with both. Here is a log using my old configuration folder.
[test-zettlr.log](https://github.com/Zettlr/Zettlr/files/6711118/test-zettlr.log)

**Comment by yaxu** [comment]
Same here in mint 20, without a config folder. File -> open dialogue doesn't work until I switch back and forth between virtual desktops.

**Comment by nathanlesage** [comment]
Aaaaallright. I just gave the whole thread here another, careful read, and it might be the following:

1. When you open a dialog, that is most of the time due to an action, e.g., a mouse click.
2. The handler which handles these starts a promise and does not exit until the dialog has been closed
3. It might be that under certain circumstances, Linux is unable to "hold back" the event in an unresolved state while at the same time displaying a modal. (I.e. it could be that Linux requires events inside a window to be handled _before_ it can be disabled.

In short, there might be a serious bug in making the dialogs modal.

***

I have just added a commit that prevents dialogs from becoming modal on Linux. Additionally, I have just initiated a manual nightly-build. **After the build is finished (~1 hour), please try again to install a nightly and see if the issue persists.**

**Comment by ajlewis2** [comment]
I'm sorry to say that the problem is still there. The "Really Delete" block opens with Delete or Cancel options, but clicking on either option does not work. ESC exits the dialog.

**Comment by ajlewis2** [comment]
Would it help if I run it using strace? It will give a very long log that I don't know how to read, but might give more clues.

**Comment by nathanlesage** [comment]
Mh, I have never worked with `strace`, so I'm not sure I can make something of this either …

I just googled \~again\~ and still didn't found much. But I did find that apparently some other folks reported similar problems with other Electron apps like VS Code and Typora.

Finally, I found that I missed the dialog that is not working, because I forgot to refactor this out 🙄

So it couldn't have worked. I have now applied the debug patch to the Delete File/Delete Dir confirmation dialogs, please wait for the next nightly to appear and then attempt again. Sorry!

**Comment by ajlewis2** [comment]
Yes, it now deletes a file. My original problem was import which does not work, but the delete dialog allows OK which deletes. :-)

**Comment by nathanlesage** [comment]
Okay, so it's all about the modal status. We're making progress here! :)

@ajlewis2 I'll re-introduce the non-modality into the Linux dialogs tomorrow or next week so that we can test all dialogs for this specific bug. If that fixes it, we know it has to do something with the fact that both the main process and the renderer process are awaiting a promise in conjunction with a dialog being modal on the main window.

It seems that there might be some bug with regard to dialogs being attached to other windows as modals, since apparently sometimes this can make the window, the dialog and/or the whole screen unresponsive.

If we can now make a reproducible example, that would be great b/c then we know if it's either Zettlr's fault or Electron's!

I don't know when I have time to write a small sample app using [Electron fiddle](https://www.electronjs.org/fiddle), but if someone of you can do this, this could speed things up. If we can prove that it's a bug in Electron, we can file an issue over there, otherwise we now have a working fix!

**Comment by yaxu** [comment]
With the most recent nightly (26th June, there isn't one for last night):
- If I click the open workspace icon in the tool bar, the dialog responds to mouse clicks.
- If I click 'open workspace' in the menus, it doesn't (unless I switch back and forth between virtual desktops)

**Comment by nathanlesage** [comment]
Yes, I have not yet managed to get that bug fix back into the versions, but will do so today. Then the next nightly (should arrive somewhere today) will contain the bug fix which should fix the dialogs.

**Comment by nathanlesage** [comment]
I have just re-implemented the debug code and triggered a new build. Once that's ready, please feel free to test all dialogs out. Now, none of these should be modal on Linux operating systems, so let's hope for the best …

**Comment by ajlewis2** [comment]
The dialogs work for, but not all the functions work. Import does a dialog, can select, but nothing is imported. Export no longer asks what sort of export I want. Ctrl-n opens a new file, but doesn't name it with the date like before. Ctrl-s opens the file browser for the first save where previously it saved in the workspace. Basically it seems the issue is fixed, but there are changes. This is with my old config file.  I'll try it letting it make a new config file. If there is a difference, I'll let you know.

**Comment by ajlewis2** [comment]
Same with new config file. I did notice that I can rename a file using context menu, but not File/Rename File.

**Comment by nathanlesage** [comment]
> Basically it seems the issue is fixed, but there are changes. This is with my old config file. I'll try it letting it make a new config file.

At this point I'm only interested in the former part of that.

Yes, I've done some changes to the app in general, and a lot of these are visual (just like with the filenames). The "File/Rename" menu is simply not yet re-instated, just as a few other things. Especially with regard to the menus there's still work to do, given that I rarely use the menu. Zettlr is still basically an alpha version, keep that in mind – for me it was just the easiest method to have me without and you with a Linux computer work together to find the root cause of that ugly bug.

Now I'll see to opening an issue towards the Electron repository …

**Comment by nathanlesage** [comment]
So, folks. Now all the menu items should work again, and due to my last changes it should now be possible that all of you can click the buttons in the dialogs again.

Would you mind confirming that real quick? Then I can for now close the issue.

**Comment by ajlewis2** [comment]
Looks good here on Mint. Delete file and directory work. Rename also. Import does not lock up, though nothing happens. But that is a different problem. Export comes up, but I get an error for export. It doesn't lock up, though. I did my test with my current config. I'll see what happens with export using a fresh one. But that again is not part of this current problem. Looks good to me.

**Comment by ajlewis2** [comment]
I removed my old config and let the nightly update Zettlr make it new. It exports and imports now. This is a great update for me with the import working.  Thanks again!

**Comment by BenjaminHerbert** [comment]
Delete **works for me** with this setup:

> Zettlr Version: 1.8.9+nightly-20210719 (UUID: d59bb5be-27f5-42bb-96be-c0bc943b50ef)
> System: linux 5.12.10-300.fc34.x86_64 (architecture: x64)

One side note: there is a shortcut <kbd>ctrl</kbd> + <kbd>&#8592;</kbd> shown in the delete dialog, this does not work. (I did not find an overlapping system shortcut.)

**Comment by nathanlesage** [comment]
> I removed my old config and let the nightly update Zettlr make it new. It exports and imports now. This is a great update for me with the import working. Thanks again!

Ouh, that indicates a problem with the config – which version of Zettlr were you using before, @ajlewis2 ?

***

@BenjaminHerbert Sounds good! With regard to the shortcut and the context actions in the editor, would you mind opening issues for these two? That would be great! Also try to find if there are errors logged in the console, which'd help me fix these more quick!

**Comment by ajlewis2** [comment]
> Ouh, that indicates a problem with the config – which version of Zettlr were you using before, @ajlewis2 ?

It was 1.8.9, same number as the nightly, only the last stable version which I got when it was presented to me upon opening the version prior to that.  If you want me to try the old config directory and get information, let me know. I can dig out the backup file I have and run with it.

**Comment by BenjaminHerbert** [comment]
@nathanlesage I openend the issues #2220 and #2221, thanks for taking a look.

**Comment by nathanlesage** [comment]
> If you want me to try the old config directory and get information, let me know. I can dig out the backup file I have and run with it.

Please do so, @ajlewis2 – since many users will update from 1.8.9 to 2.0.0, and if their configs throw weird errors, that would be not ideal! So the more bugs in that regard we can fix, the better!

**Comment by ajlewis2** [comment]
> Please do so, @ajlewis2 – since many users will update from 1.8.9 to 2.0.0, and if their configs throw weird errors, that would be not ideal! So the more bugs in that regard we can fix, the better!

Using Export button. Choose HTML to current directory. I get window open with error message:
An error occurred on export: Unknown option --defaults.
Unknown option --defaults.
Try pandoc --help for more information.
============
I removed the config directory and installed the old version of program. I ran the program to produce a new config directory. I tried the old program to see the problems that you solved with the new one. Now I have a fresh config directory.

I installed nightly and opened it with the fresh config from the old program. The import and export worked flawlessly. 

I recall there was a problem that was corrected by changing a line in the Export section for pandoc. I will post that line here where I think the problem might be. I will also post a screenshot of how the Export config window looked in the old version so you can see where that line was put. I don't see the line in the new version of the program.

pandoc "$infile$" -f markdown $outflag$ $tpl$ $toc$ $tocdepth$ $bibliography$ $cslstyle$ $standalone$ --pdf-engine=xelatex --mathjax -o "$outfile$"

![zettrl-old-pref](https://user-images.githubusercontent.com/67911821/126867415-03492106-e2dd-440b-af5e-9c82eee24f45.png)

**Comment by ajlewis2** [comment]
> Please do so, @ajlewis2 – since many users will update from 1.8.9 to 2.0.0, and if their configs throw weird errors, that would be not ideal! So the more bugs in that regard we can fix, the better!

The problem is setting to not use the internal pandoc for exports. I turned this off and I get the error on export and import doesn't work. I put my original config file back in place, and did the following:
Try export: get error
Set in Export preference to use internal pandoc
Try export: get error
Restart program
Try export: works.
Try import: That does not work, so there is something else involved with that, I guess.

Do you have an idea of what I might try to make my old config file work for import with the new program? I'll test it out.

**Comment by nathanlesage** [comment]
> The problem is setting to not use the internal pandoc for exports. I turned this off and I get the error on export and import doesn't work.

That is because this setting is only applied during the application boot, so if you change it, you have to restart the app. I think I should add a "restart required" to that as well …

**Comment by nathanlesage** [comment]
So I'm closing since the issue seems to have been mitigated after I made all dialogs non-modal on Linux. If that is _not_ right, please just ping me in this thread.

(I'm currently hoping to close off a few issues, please bear with me, my "open-issue-anxiety" is back …)

**Comment by nathanlesage** [comment]
> I am running 1.8.9 on an appimage on Ubuntu 20.04 and I still see this issue.

Right, 1.8.9 obviously doesn't work. It's been fixed in the current 2.0 betas.
